os/hosting-cloud-config-using-nginx.md

```diff
@@ -0,0 +1,21 @@
+## Hosting Cloud-Config Using Nginx
+
+One of the ways to host your `cloud-config` files is to use [nginx][nginx]. You can also use [http_sub_module][http_sub_module] which will allow you to use `$private_ipv4` and `$public_ipv4` substitution variables referenced in other documents. By default this module is enabled in official nginx packages and in most Linux distributions.
+
+Here is an example nginx configuration which will substitute `$public_ipv4` and `$private_ipv4` (depends on your nginx server location and NAT configuration) variables:
```

Comment:
The example nginx configuration below will replace the `$public_ipv4` and `$private_ipv4` variables for each client connection from a CoreOS machine booting through the cloud-init process.